src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -723,8 +723,7 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded)
 
             private void RecordAssemblySymbols(Compilation compilation, Dictionary<MetadataReference, ProjectId> metadataReferenceToProjectId)
             {
-                // TODO: Record source assembly to project mapping
-                // RecordSourceOfAssemblySymbol(compilation.Assembly, this.ProjectState.Id);
+                RecordSourceOfAssemblySymbol(compilation.Assembly, this.ProjectState.Id);
```

Comment:
Why it was commented out before?

Reply:
It's really hard to tell.  @jasonmalinowski and i had a long conversation today trying to pool our best knowledge about the historyof things here and we really don't have a good idea about why things are done this way.  

I think it may have just been bad code, and we were too worried to change it.

Reply:
So unless @dpoeschl remembers at all, but it looks like it was TODO'ed when this originally introduced in #10663.

I had a random thought (or perhaps a vague recollection?) which is there _could_ be an issue where you can take a symbol, get to it's IAssemblySymbol, and this map could have the project, _but_ if the Compilation has already GC'ed then the compilation you fetch would have a new IAssemblySymbol. I would have assumed that a Compilation and SourceAssemblySymbol root each other, but maybe not...? If that _is_ the problem then it still might be more sane to just still use this map, and in GetProjectState just call TryGetCompilation and verify it after you've looked in the map; at least that way it's not O(n)...